Source/cmMakefileTargetGenerator.cxx

```diff
@@ -1748,11 +1748,50 @@ void cmMakefileTargetGenerator::AppendObjectDepends(
                                          this->BuildFileNameFull.c_str());
 }
 
+void cmMakefileTargetGenerator::AppendLinkDependsForJson(
+        std::vector<std::string>& depends, const std::string& linkLanguage)
+{
+    // Add dependencies on the compiled object files.
+    std::string const& fullPath =
+            this->LocalGenerator->ConvertToFullPath(
+                    this->LocalGenerator->GetHomeRelativeOutputPath());
+    for (std::string const& obj : this->Objects) {
+        std::string objTarget = cmStrCat(fullPath, obj);
```

Comment:
Is this always the right behaviour? I think fullPath can be changed according to the working directory from where the object is built. I am not sure it is possible in cmake, however this may be a source of issue with lapack

Reply:
fixed